viewing_party/party.py

```diff
@@ -1,21 +1,125 @@
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title and genre and rating:
+        movie_dict.update({"title": title})
+        movie_dict.update({"genre": genre})
+        movie_dict.update({"rating": rating})
+        return movie_dict
+    else:
+        return None   
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+    
+
+def watch_movie(user_data, title):
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == title:
+            user_data["watched"].append(movie_dict)
+            user_data["watchlist"].remove(movie_dict)
+            return user_data
+    return user_data
+    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_list = []
+    rating_total = 0
+    
+    for movie in user_data["watched"]:
+        if len(user_data["watched"]) == 0:
+            movie["rating"] = 0.0

```

Comment:
hmm  what is this doing? If there are no movies in "watched", I don't think we can then add a key to a non-existent dictionary. I think this might error out. Let's remove these lines.

Also, we don't want to be making modifications to the user's data *unless* that's the purpose of the function. This function `get_watched_avg_rating` is only supposed to make calculations and return them.